RecDP/pyrecdp/primitives/llmutils/document/reader.py

```diff
@@ -119,79 +136,171 @@ def load_data(self, file: Path, **load_kwargs: Any) -> List[Document]:
-            docs: List[Document] = list()
-            for element in elements:
-                metadata = self._get_metadata(path)
-                # NOTE(MthwRobinson) - the attribute check is for backward compatibility
-                # with unstructured<0.4.9. The metadata attributed was added in 0.4.9.
-                if hasattr(element, "metadata"):
-                    metadata.update(element.metadata.to_dict())
-                if hasattr(element, "category"):
-                    metadata["category"] = element.category
-                docs.append(Document(text=str(element), metadata=metadata))
-        elif self.mode == "paged":
-            text_dict: Dict[int, str] = {}
-            meta_dict: Dict[int, Dict] = {}
-
-            for idx, element in enumerate(elements):
-                metadata = self._get_metadata(path)
-                if hasattr(element, "metadata"):
-                    metadata.update(element.metadata.to_dict())
-                page_number = metadata.get("page_number", 1)
-
-                # Check if this page_number already exists in docs_dict
-                if page_number not in text_dict:
-                    # If not, create new entry with initial text and metadata
-                    text_dict[page_number] = str(element) + "\n\n"
-                    meta_dict[page_number] = metadata
+            file_suffix = input_file.suffix.lower()
+            if file_suffix in self.supported_suffix:
+                if file_suffix in CUSTOMIZE_SUPPORTED_SUFFIX:
+                    if file_suffix not in self.file_extractor:
+                        file_base_reader_cls: Type[FileBaseReader] = CUSTOMIZE_SUPPORTED_SUFFIX[file_suffix]
+                        self.file_extractor[file_suffix] = file_base_reader_cls(
+                            input_file,
+                            single_text_per_document=self.single_text_per_document
+                        )
+                    loader = self.file_extractor[file_suffix]
+                    return loader.load()
                 else:
-                    # If exists, append to text and update the metadata
-                    text_dict[page_number] += str(element) + "\n\n"
-                    meta_dict[page_number].update(metadata)
-
-            # Convert the dict to a list of Document objects
-            docs = [
-                Document(text=text_dict[key], metadata=meta_dict[key])
-                for key in text_dict.keys()
-            ]
-        elif self.mode == "single":
-            metadata = self._get_metadata(path)
-            text = "\n\n".join([str(el) for el in elements])
-            docs = [Document(text=text, metadata=metadata)]
-        else:
-            raise ValueError(f"mode of {self.mode} not supported.")
-        return docs
+                    from pyrecdp.core.import_utils import import_langchain
+                    import_langchain()
+                    from langchain.document_loaders import UnstructuredFileLoader
```

Comment:
@chaojun-zhang , Why we will have two places for impl? 
Can we merge document/reader.py to doc_loader?

Reply:
The DirectoryLoader class in doc_loader is an operator that uses the DirectoryReader class in document/reader.py. The DirectoryReader class is also used by the DocumentExtractor class.